gdal/ogr/ogrsf_frmts/cad/libopencad/dwg/r2000.cpp

```diff
@@ -726,13 +770,15 @@ int DWGFileR2000::CreateFileMap()
         if( dSectionSize == 2 )
             break; // last section is empty.
 
-        char * pabySectionContent  = new char[dSectionSize + 4];
-        size_t nBitOffsetFromStart = 0;
+        char * pabySectionContent  = new char[2 + dSectionSize];
         size_t nRecordsInSection   = 0;
 
         // read section data
-        pFileIO->Read( pabySectionContent, dSectionSize );
-        unsigned int dSectionBitSize = (dSectionSize * 8) - 128; // 8 + 8*7
+        pFileIO->Seek( -2L, CADFileIO::SeekOrigin::CUR );
+        pFileIO->Read( pabySectionContent, 2 + dSectionSize );
+        unsigned int dSectionBitSize = (2 + dSectionSize - 2) * 8;
```

Comment:
Section structure is awkward, so I left this as a reminder. Comment my be more appropriate.